stream/src/main/scala/org/apache/pekko/stream/javadsl/Sink.scala

```diff
@@ -313,6 +313,27 @@ object Sink {
       scaladsl.Sink.seq[In].mapMaterializedValue(fut => fut.map(sq => sq.asJava)(ExecutionContext.parasitic).asJava))
   }
 
+  /**
+   * A `Sink` that counts all incoming elements until upstream terminates.
+   *
+   * Since upstream may be unbounded, consider using `Flow[T].take` or the stricter `Flow[T].limit`
+   * (and their variants) to ensure boundedness. The sink materializes into a `CompletionStage` of `Long`
+   * containing the total count of elements that passed through.
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Backpressures when''' never (counting is a lightweight operation)
+   *
+   * '''Cancels when''' never
+   *
+   * @return a `Sink` that materializes to a `CompletionStage[Long]` with the element count
+   * @since 2.0.0
+   *
+   * See also [[Flow.limit]], [[Flow.limitWeighted]], [[Flow.take]], [[Flow.takeWithin]], [[Flow.takeWhile]]
+   */
+  def count[In]: Sink[In, CompletionStage[java.lang.Long]] = new Sink(
+    scaladsl.Sink.count[In].mapMaterializedValue(_.mapTo[java.lang.Long].asJava))
```

Comment:
Same as the reactor core, but this conversion looks bad. Is there a better way?

Reply:
This is what looks natural to me

Reply:
Why scala compiler can't do this for me？

Reply:
Not on laptop, I'll check when I get t back home

Reply:
Not sure why A Scala.Long can not be automaticly convert to a Java java.lang.Long, if this is the case, then there may be some `Int` `Long` leak to java api, I have not grep that, but I think so.